### PR TITLE
Add ampliconsuite recipe

### PR DESCRIPTION
**Pull request by jluebeck**
This PR adds are recipe for AmpliconSuite, a collection of bioinformatic command-line tools for studying focal genomic amplifications.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by jluebeck** [comment]
@BiocondaBot please add label

**Comment by peterpru** [comment]
@BiocondaBot please fetch artifacts

File: recipes/ampliconsuite/build.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# pull source code for AA and move scripts into a library
+TARGET="AmpliconArchitect"
+TARGET_VERSION="1.3.r5"
+wget https://github.com/AmpliconSuite/${TARGET}/archive/refs/tags/v${TARGET_VERSION}.zip
```
Comment (bernt-matthias) [inline recipes/ampliconsuite/build.sh:6]: IMO this should be realized as additional sources in meta.yaml, and adding `__init__.py files as patches.

File: recipes/ampliconsuite/build.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
```
Comment (bernt-matthias) [inline recipes/ampliconsuite/build.sh:2]:
```suggestion

set -ex

```

File: recipes/ampliconsuite/build.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+cp -r ${TARGET}-${TARGET_VERSION}/resources/ ampliconclassifierlib/resources/
+touch ampliconclassifierlib/__init__.py
+rm v${TARGET_VERSION}.zip
+
+# make the bin dir if it doesn't exist
+mkdir -p $PREFIX/bin
+
+# copy driver scripts
+cp AmpliconSuite-pipeline.py ${PREFIX}/bin/AmpliconSuite-pipeline.py
+cp GroupedAnalysisAmpSuite.py ${PREFIX}/bin/GroupedAnalysisAmpSuite.py
```
Comment (bernt-matthias) [inline recipes/ampliconsuite/build.sh:30]: Maybe in the future the executables could be defined via setup.py

File: recipes/ampliconsuite/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    version: {{ version }}
+
+source:
+   url: https://github.com/AmpliconSuite/AmpliconSuite-pipeline/archive/v{{ version }}.tar.gz
+   sha256: 129a2606a7e66168925784205a75ff6134c2280e2379ee523375ea6aa3a384f3
+
+build:
+    noarch: python
+    number: 0
+    skip: False
```
Comment (bernt-matthias) [inline recipes/ampliconsuite/meta.yaml:14]:
```suggestion
```

File: recipes/ampliconsuite/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+    build:
+        - python
```
Comment (bernt-matthias) [inline recipes/ampliconsuite/meta.yaml:18]: python >=3 ?

File: recipes/ampliconsuite/build.sh (hunk 1 above)
Comment (bernt-matthias) [reply recipes/ampliconsuite/build.sh:6]: If amplicon architect and classifier are also of interest stand alone a separate recipe might also be a good option.

Comment (jluebeck) [reply recipes/ampliconsuite/build.sh:6]: Hi Matthias - thanks so much for reviewing this PR. Regarding your second comment - we would prefer if most users not attempt to use the AmpliconArchitect and AmpliconClassifier modules in a standalone fashion. The functionality from both can be accessed from `AmpliconSuite-pipeline.py`. That is the single entrypoint we would like to support going forward, as it captures our best practices for using the encompassed modules.

I will update the recipe with your excellent suggestions and get back to you shortly.

Thanks again,
Jens

**Comment by jluebeck** [comment]
Hi @bernt-matthias, thanks again for these suggestions. I have incorporated all of them into the recent commits. Local build, local install and tests looked good to me. Mind taking a look at this PR when you have a moment?

Cheers,
Jens

**Comment by jluebeck** [comment]
@BiocondaBot please fetch artifacts

**Review by bernt-matthias — APPROVED** [review]
Hey @jluebeck. Have you been able to test this with the artifacts?

**Comment by jluebeck** [comment]
@BiocondaBot please fetch artifacts

**Comment by jluebeck** [comment]
@BiocondaBot please fetch artifacts

**Comment by jluebeck** [comment]
Hi @bernt-matthias, I had to fix one pathing issue in the `build.sh` script. Following that, I have downloaded and tested the very latest artifact on a test dataset and not experienced any errors during the process.

Thanks,
Jens

**Comment by bernt-matthias** [comment]
Thanks @jluebeck

**Pull request merged**